scripts/check-deps/check-dependencies.js

```diff
@@ -0,0 +1,319 @@
+		const modules = new Set();
+
+		print(basename(`  ${name}: `));
+
+		await forEachSourceFile(name, async source => {
+			const contents = await readFileAsync(source);
+			const ast = parse(contents.toString(), {
+				plugins: ['flow'],
+				sourceType: 'unambiguous',
+			});
+			traverse(ast, {
+				CallExpression(path) {
+					const callee = path.get('callee');
+					if (callee.isIdentifier({name: 'require'})) {
+						const args = path.get('arguments');
+						if (args.length && args[0].isStringLiteral()) {
+							recordDependency(args[0].node.value, modules);
+						}
+					}
+				},
+				ImportDeclaration(path) {
+					// `importKind` may be "value" or "type".
+					if (path.get('importKind').node === 'value') {
+						const source = path.get('source');
+						if (source.isStringLiteral()) {
+							recordDependency(source.node.value, modules);
+						}
+					}
+				},
+			});
+		});
+
+		const dependencies = new Set([
+			...Object.keys(config.dependencies || {}),
+			...Object.keys(config.peerDependencies || {}),
+		]);
+		for (const moduleName of modules) {
+			if (!dependencies.has(moduleName)) {
+				if (!missing[name]) {
+					missing[name] = [];
+				}
+				missing[name].push(moduleName);
+			}
+		}
+
+		if (missing[name] && missing[name].length) {
+			print.line.red('MISSING');
+			missing[name].forEach(dependency =>
+				print.line(`    ${dependency}`)
+			);
+		} else {
+			print.line.green('OK');
+		}
+	}, WORKSPACE_PACKAGES_WHITELIST);
+
+	print();
+	const success = Object.keys(missing).length === 0;
+	if (!success) {
+		print.line.red('Add missing dependencies with:\n');
+		Object.entries(missing).forEach(([name, dependencies]) => {
+			print.line.red(
+				`(cd packages/${name} && yarn add ${dependencies.join(' ')})`
+			);
+		});
+		print();
+	}
+	return success;
+}
+
+/**
+ * Make sure packages require identical versions of common dependencies.
+ */
+async function checkForMismatchedDependencyVersions() {
+	print.line.yellow('Checking for mismatched dependency versions:\n');
+	const registry = {};
+
+	let success = true;
+	await forEachPackage((name, config) => {
+		const dependencies = getDependencies(config);
+		dependencies.forEach(([dependency, version]) => {
+			if (!registry[dependency]) {
+				registry[dependency] = {};
+			}
+			if (!registry[dependency][version]) {
+				registry[dependency][version] = new Set();
+			}
+			recordDependency(name, registry[dependency][version]);
+		});
+	}, WORKSPACE_PACKAGES_WHITELIST);
+
+	for (const [dependency, versions] of Object.entries(registry)) {
+		print(`  ${dependency}:  `);
+		if (Object.keys(versions).length === 1) {
+			print.line.green('OK');
+		} else {
+			success = false;
+			print.line.red('BAD');
+			for (const [version, dependees] of Object.entries(versions)) {
+				for (const dependee of dependees) {
+					print.line.red(`    ${dependee} -> ${version}`);
+				}
+			}
+		}
+	}
+
+	print();
+	return success;
+}
+
+/**
+ * Can be used if
```

Comment:
If what?

Reply:
I forgot it, I gonna update this description. 😅